frontend/src/lib/components/apps/components/buttons/AppButton.svelte

```diff
@@ -108,22 +112,30 @@
 		event?.stopPropagation()
 		event?.preventDefault()
 
-		$selectedComponent = [id]
-		const inputOutput = { result: outputs.result.peak(), loading: true }
-		if (rowContext && rowInputs) {
-			rowInputs.set(id, inputOutput)
-		}
-		if (iterContext && listInputs) {
-			listInputs.set(id, inputOutput)
-		}
-		if (preclickAction) {
-			await preclickAction()
+		const action = async () => {
+			$selectedComponent = [id]
```

Comment:
this should probably be also above the action, the button should be selected as soon as it's clicked. But makes sense to also have it within the deferred action